### PR TITLE
New rule: unnecessary_override

### PR DESCRIPTION
**Pull request by JPaulsen**

### CODE REVIEW

**Comment by JPaulsen** [comment]
23 lints in the sdk, no false positives.

File: lib/src/rules/unnecessary_override.dart [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  }
+
+  @override
+  visitExpressionStatement(ExpressionStatement node) {
+    node.expression.accept(this);
+  }
+
+  @override
+  visitMethodDeclaration(MethodDeclaration node) {
+    if (!node.isGetter && !node.isSetter) {
```
Comment (bwilkerson) [inline lib/src/rules/unnecessary_override.dart]: Why are getters and setters excluded from being linted?

File: test/rules/unnecessary_override.dart [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  int getB(Iterable a) { // LINT
+    return super.getB(a);
+  }
+
+  @override
+  int getC(Iterable a) { // OK
+    print("something");
+    return super.getC(a);
+  }
+}
```
Comment (bwilkerson) [inline test/rules/unnecessary_override.dart]: This needs more tests. What about overriding a method with more than one parameter, both with and without changing the order of the arguments?
```
int max(int a, int b) => super.max(a, b); // LINT
int max(int a, int b) => super.max(b, a); // OK
```
Or a method with positional optional parameters, or named parameters (where the order of the named parameters can be changed, both with and without making the override ok)?
```
int m({int a, int b}) => super.m(a: a, b: b); // LINT
int m({int a, int b}) => super.m(b: b, a: a); // LINT
int m({int a, int b}) => super.m(b: a, a: b); // OK
```
Or an operator:
```
int operator +(int other) => super + other; // LINT
```

File: lib/src/rules/unnecessary_override.dart (hunk 1 above)
Comment (JPaulsen) [inline lib/src/rules/unnecessary_override.dart]: Fixed

File: test/rules/unnecessary_override.dart (hunk 2 above)
Comment (JPaulsen) [inline test/rules/unnecessary_override.dart]: Fixed

**Comment by JPaulsen** [comment]
I added the case with no targets in getters and setters and also added tests

**Pull request merged**